## Part 1: Explore the Data

Import the data and use Pandas to learn more about the dataset.

In [76]:
import pandas as pd

src_data_path = r'Resources/client_dataset.csv'
df = pd.read_csv(src_data_path)
df.head(10)

first      last                                           job  \
0     Donald   Harding                                  Immunologist   
1    Tiffany     Myers                               Music therapist   
2    Shannon    Watson                                  Immunologist   
3     Nathan     Baker                         Accounting technician   
4  Christina  Schwartz                                  Chiropractor   
5    Jessica     Reyes  Production designer, theatre/television/film   
6      Brian     Moore                                       Barista   
7      Jesus    Wright                            Theme park manager   
8    Rebecca     Brown                                   Chiropodist   
9       Greg    Garcia                            Secretary, company   

                 phone                                email  client_id  \
0   793-904-7725x39308     harding.donald.7185@sullivan.com      58515   
1     201.442.4543x942    myers.t.6537@ferguson-johnson.net      37609   
2    687.737.9424x8503                swatson8146@payne.net      57113   
3     827-788-8123x012               bakernathan@benson.com      46554   
4         265-829-3643  christinaschwartz9252@mcconnell.com      92089   
5           5016915923             jessica.reyes@grimes.net      33615   
6  +1-893-400-2733x982          moore.official@martinez.com      87213   
7   605.947.9467x66513        wright.jesus.7579@meadows.com      46709   
8     665-955-0156x926                 rbrown850@lewis.info      73362   
9   (231)342-7216x1421                  ggarcia@bentley.biz      58789   

   order_id  order_date  order_week  order_year         item_id     category  \
0   8953482  2023-04-28          17        2023  EUD29711-63-6U        decor   
1   8069089  2023-05-19          20        2023  XDA18116-89-4A  consumables   
2   1902144  2023-01-29           4        2023  ABE59463-05-7E     software   
3   9031802  2023-04-25          17        2023  ZMM00836-65-0C  consumables   
4   1322274  2023-05-28          21        2023  BZX55559-12-3X  consumables   
5   1327730  2023-05-02          18        2023  BMB23158-88-9A        decor   
6   8080280  2023-04-19          16        2023  UAC50632-25-4E  consumables   
7   8946782  2023-02-14           7        2023  UEU25316-01-1U  consumables   
8   9394118  2023-05-13          19        2023  DDA06248-98-7Z  consumables   
9   9665852  2023-04-14          15        2023  DUU85042-87-5E    furniture   

          subcategory  unit_price  unit_cost  unit_weight  qty  line_number  
0            wall art     1096.80     762.71         7.50  105            1  
1                pens       24.95      15.09         1.49   21            0  
2  project management       13.52       7.86         1.68   39            6  
3                pens       36.42      24.85         1.23   29            3  
4                misc      195.10     108.17        46.43   20            1  
5              plants      172.84     154.44         2.79   78            0  
6    kitchen supplies      245.56     231.95         1.73   56            5  
7                pens       20.99      16.97         5.12   22            4  
8      paper products       83.86      59.00         0.27   49            3  
9             seating       66.88      54.15         2.47   78            1

In [77]:
# View the column names in the data
df.columns

Index(['first', 'last', 'job', 'phone', 'email', 'client_id', 'order_id',
       'order_date', 'order_week', 'order_year', 'item_id', 'category',
       'subcategory', 'unit_price', 'unit_cost', 'unit_weight', 'qty',
       'line_number'],
      dtype='object')

In [5]:
# Use the describe function to gather some basic statistics
df.describe


<bound method NDFrame.describe of            first       last                      job               phone  \
0         Donald    Harding             Immunologist  793-904-7725x39308   
1        Tiffany      Myers          Music therapist    201.442.4543x942   
2        Shannon     Watson             Immunologist   687.737.9424x8503   
3         Nathan      Baker    Accounting technician    827-788-8123x012   
4      Christina   Schwartz             Chiropractor        265-829-3643   
...          ...        ...                      ...                 ...   
54634     Lauren      Reese           Radio producer  977.877.5272x11382   
54635    Derrick      Moore                 Musician        358.661.5483   
54636     Monica  Gutierrez         Graphic designer    294.805.9100x339   
54637      Wanda    Solomon             Toxicologist       (311)767-4924   
54638     Dwayne       Bell  Chief Financial Officer   606-619-7667x9508   

                                     email  client_id

#### Use this space to do any additional research and familiarize yourself with the data.

- Overall Info About the Client and their Purchase habits

In [6]:
client_info = df[['client_id','first', 'last', 'phone', 'email', 
                          'job', 'order_id', 'order_date', 'order_week',  'item_id', 'category', 'subcategory', 'unit_price', 
                          'unit_cost', 'unit_weight', 'qty', 'line_number']]
client_info

client_id      first       last               phone  \
0          58515     Donald    Harding  793-904-7725x39308   
1          37609    Tiffany      Myers    201.442.4543x942   
2          57113    Shannon     Watson   687.737.9424x8503   
3          46554     Nathan      Baker    827-788-8123x012   
4          92089  Christina   Schwartz        265-829-3643   
...          ...        ...        ...                 ...   
54634      29043     Lauren      Reese  977.877.5272x11382   
54635      41908    Derrick      Moore        358.661.5483   
54636      35176     Monica  Gutierrez    294.805.9100x339   
54637      24485      Wanda    Solomon       (311)767-4924   
54638      24007     Dwayne       Bell   606-619-7667x9508   

                                     email                      job  order_id  \
0         harding.donald.7185@sullivan.com             Immunologist   8953482   
1        myers.t.6537@ferguson-johnson.net          Music therapist   8069089   
2                    swatson8146@payne.net             Immunologist   1902144   
3                   bakernathan@benson.com    Accounting technician   9031802   
4      christinaschwartz9252@mcconnell.com             Chiropractor   1322274   
...                                    ...                      ...       ...   
54634   late_reese_4081@montoya-chavez.org           Radio producer   9021716   
54635         derrick.moore.2602@pope.info                 Musician   6290153   
54636            gutierrezm3195@morris.org         Graphic designer   8692622   
54637            solomonwanda5962@ross.org             Toxicologist   7592730   
54638                     belld@galvan.com  Chief Financial Officer   7489403   

       order_date  order_week         item_id     category  \
0      2023-04-28          17  EUD29711-63-6U        decor   
1      2023-05-19          20  XDA18116-89-4A  consumables   
2      2023-01-29           4  ABE59463-05-7E     software   
3      2023-04-25          17  ZMM00836-65-0C  consumables   
4      2023-05-28          21  BZX55559-12-3X  consumables   
...           ...         ...             ...          ...   
54634  2023-04-26          17  UCA28532-31-0B     software   
54635  2023-02-08           6  EBD29277-23-7U  consumables   
54636  2023-03-05           9  EBD30093-55-9Z  consumables   
54637  2023-01-18           3  XMZ38239-49-3D    furniture   
54638  2023-01-04           1  ZCZ05589-50-5D     software   

              subcategory  unit_price  unit_cost  unit_weight   qty  \
0                wall art     1096.80     762.71         7.50   105   
1                    pens       24.95      15.09         1.49    21   
2      project management       13.52       7.86         1.68    39   
3                    pens       36.42      24.85         1.23    29   
4                    misc      195.10     108.17        46.43    20   
...                   ...         ...        ...          ...   ...   
54634           logistics       83.13      51.60         2.25    33   
54635    kitchen supplies      206.59     175.46        11.70    47   
54636   bathroom supplies       65.66      57.31         4.16   475   
54637              stands        1.48       1.22        18.04   112   
54638  project management        3.01       1.61         2.07  1031   

       line_number  
0                1  
1                0  
2                6  
3                3  
4                1  
...            ...  
54634            8  
54635            0  
54636            6  
54637            0  
54638            5  

[54639 rows x 17 columns]

- Client Personal Identifiable Info

In [7]:

client_pii = df[['client_id','first', 'last', 'phone', 'email', 'job']]
client_pii


client_id      first       last               phone  \
0          58515     Donald    Harding  793-904-7725x39308   
1          37609    Tiffany      Myers    201.442.4543x942   
2          57113    Shannon     Watson   687.737.9424x8503   
3          46554     Nathan      Baker    827-788-8123x012   
4          92089  Christina   Schwartz        265-829-3643   
...          ...        ...        ...                 ...   
54634      29043     Lauren      Reese  977.877.5272x11382   
54635      41908    Derrick      Moore        358.661.5483   
54636      35176     Monica  Gutierrez    294.805.9100x339   
54637      24485      Wanda    Solomon       (311)767-4924   
54638      24007     Dwayne       Bell   606-619-7667x9508   

                                     email                      job  
0         harding.donald.7185@sullivan.com             Immunologist  
1        myers.t.6537@ferguson-johnson.net          Music therapist  
2                    swatson8146@payne.net             Immunologist  
3                   bakernathan@benson.com    Accounting technician  
4      christinaschwartz9252@mcconnell.com             Chiropractor  
...                                    ...                      ...  
54634   late_reese_4081@montoya-chavez.org           Radio producer  
54635         derrick.moore.2602@pope.info                 Musician  
54636            gutierrezm3195@morris.org         Graphic designer  
54637            solomonwanda5962@ross.org             Toxicologist  
54638                     belld@galvan.com  Chief Financial Officer  

[54639 rows x 6 columns]

- Client Order Info

In [8]:

order_info = df[['order_id', 'order_date', 'order_week','unit_price', 
                          'unit_cost', 'unit_weight', 'qty']]
order_info

order_id  order_date  order_week  unit_price  unit_cost  unit_weight  \
0       8953482  2023-04-28          17     1096.80     762.71         7.50   
1       8069089  2023-05-19          20       24.95      15.09         1.49   
2       1902144  2023-01-29           4       13.52       7.86         1.68   
3       9031802  2023-04-25          17       36.42      24.85         1.23   
4       1322274  2023-05-28          21      195.10     108.17        46.43   
...         ...         ...         ...         ...        ...          ...   
54634   9021716  2023-04-26          17       83.13      51.60         2.25   
54635   6290153  2023-02-08           6      206.59     175.46        11.70   
54636   8692622  2023-03-05           9       65.66      57.31         4.16   
54637   7592730  2023-01-18           3        1.48       1.22        18.04   
54638   7489403  2023-01-04           1        3.01       1.61         2.07   

        qty  
0       105  
1        21  
2        39  
3        29  
4        20  
...     ...  
54634    33  
54635    47  
54636   475  
54637   112  
54638  1031  

[54639 rows x 7 columns]

- Product Information

In [9]:
item_info = df[['line_number', 'item_id', 'category', 'subcategory']]
item_info




line_number         item_id     category         subcategory
0                1  EUD29711-63-6U        decor            wall art
1                0  XDA18116-89-4A  consumables                pens
2                6  ABE59463-05-7E     software  project management
3                3  ZMM00836-65-0C  consumables                pens
4                1  BZX55559-12-3X  consumables                misc
...            ...             ...          ...                 ...
54634            8  UCA28532-31-0B     software           logistics
54635            0  EBD29277-23-7U  consumables    kitchen supplies
54636            6  EBD30093-55-9Z  consumables   bathroom supplies
54637            0  XMZ38239-49-3D    furniture              stands
54638            5  ZCZ05589-50-5D     software  project management

[54639 rows x 4 columns]

In [10]:
# What three item categories had the most entries?
most_cat_entries = item_info['category'].value_counts().head(3)
most_cat_entries

category
consumables    23538
furniture      11915
software        8400
Name: count, dtype: int64

In [33]:
# For the category with the most entries, which subcategory had the most entries?

# most_sub_cat_entries = pd.concat(most_cat_entries, item_info['subcategory'])
most_subcat_entries = item_info['subcategory'].value_counts().head(1)
most_subcat_entries

subcategory
bathroom supplies    6424
Name: count, dtype: int64

In [31]:
# Which five clients had the most entries in the data?
most_five_client_entries = client_pii['client_id'].value_counts().head(5)
most_five_client_entries

client_id
33615    220
66037    211
46820    209
38378    207
24741    207
Name: count, dtype: int64

In [38]:
# Store the client ids of those top 5 clients in a list.
list_top_fiveclients = list(most_five_client_entries.index)
list_top_fiveclients

[33615, 66037, 46820, 38378, 24741]

In [72]:
# How many total units (the qty column) did the client with the most entries order order?
clientID_Max = client_pii['client_id'].value_counts().idxmax()
clientID_Max
clientID = df[client_pii['client_id'] == clientID_Max]
total_qty = clientID['qty'].sum()
print(f'Client ID {clientID_Max} ordered the most items which sums up to {total_qty} quantities')


Client ID 33615 ordered the most items which sums up to 64313 quantities


## Part 2: Transform the Data
Do we know that this client spent the more money than client 66037? If not, how would we find out? Transform the data using the steps below to prepare it for analysis.

In [81]:
# Create a column that calculates the subtotal for each line using the unit_price and the qty
df1 = df.copy()
df1['subtotal'] = order_info['unit_price'] * order_info['qty']
df1

first       last                      job               phone  \
0         Donald    Harding             Immunologist  793-904-7725x39308   
1        Tiffany      Myers          Music therapist    201.442.4543x942   
2        Shannon     Watson             Immunologist   687.737.9424x8503   
3         Nathan      Baker    Accounting technician    827-788-8123x012   
4      Christina   Schwartz             Chiropractor        265-829-3643   
...          ...        ...                      ...                 ...   
54634     Lauren      Reese           Radio producer  977.877.5272x11382   
54635    Derrick      Moore                 Musician        358.661.5483   
54636     Monica  Gutierrez         Graphic designer    294.805.9100x339   
54637      Wanda    Solomon             Toxicologist       (311)767-4924   
54638     Dwayne       Bell  Chief Financial Officer   606-619-7667x9508   

                                     email  client_id  order_id  order_date  \
0         harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1        myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                    swatson8146@payne.net      57113   1902144  2023-01-29   
3                   bakernathan@benson.com      46554   9031802  2023-04-25   
4      christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   
...                                    ...        ...       ...         ...   
54634   late_reese_4081@montoya-chavez.org      29043   9021716  2023-04-26   
54635         derrick.moore.2602@pope.info      41908   6290153  2023-02-08   
54636            gutierrezm3195@morris.org      35176   8692622  2023-03-05   
54637            solomonwanda5962@ross.org      24485   7592730  2023-01-18   
54638                     belld@galvan.com      24007   7489403  2023-01-04   

       order_week  order_year         item_id     category  \
0              17        2023  EUD29711-63-6U        decor   
1              20        2023  XDA18116-89-4A  consumables   
2               4        2023  ABE59463-05-7E     software   
3              17        2023  ZMM00836-65-0C  consumables   
4              21        2023  BZX55559-12-3X  consumables   
...           ...         ...             ...          ...   
54634          17        2023  UCA28532-31-0B     software   
54635           6        2023  EBD29277-23-7U  consumables   
54636           9        2023  EBD30093-55-9Z  consumables   
54637           3        2023  XMZ38239-49-3D    furniture   
54638           1        2023  ZCZ05589-50-5D     software   

              subcategory  unit_price  unit_cost  unit_weight   qty  \
0                wall art     1096.80     762.71         7.50   105   
1                    pens       24.95      15.09         1.49    21   
2      project management       13.52       7.86         1.68    39   
3                    pens       36.42      24.85         1.23    29   
4                    misc      195.10     108.17        46.43    20   
...                   ...         ...        ...          ...   ...   
54634           logistics       83.13      51.60         2.25    33   
54635    kitchen supplies      206.59     175.46        11.70    47   
54636   bathroom supplies       65.66      57.31         4.16   475   
54637              stands        1.48       1.22        18.04   112   
54638  project management        3.01       1.61         2.07  1031   

       line_number   subtotal  
0                1  115164.00  
1                0     523.95  
2                6     527.28  
3                3    1056.18  
4                1    3902.00  
...            ...        ...  
54634            8    2743.29  
54635            0    9709.73  
54636            6   31188.50  
54637            0     165.76  
54638            5    3103.31  

[54639 rows x 19 columns]

In [146]:
# Create a column for shipping price.
# Assume a shipping price of $7 per pound for orders over 50 pounds and $10 per pound for items 50 pounds or under.
def assign_shipping_prices(item_weight):
    match item_weight:
        case num if num > 50:
            item_weight = item_weight * 7
        case num if num <= 50:
            item_weight = item_weight * 10
    return item_weight

df1['shipping_price'] = order_info['unit_weight'].apply(assign_shipping_prices)
print(df1.describe)


<bound method NDFrame.describe of            first       last                      job               phone  \
0         Donald    Harding             Immunologist  793-904-7725x39308   
1        Tiffany      Myers          Music therapist    201.442.4543x942   
2        Shannon     Watson             Immunologist   687.737.9424x8503   
3         Nathan      Baker    Accounting technician    827-788-8123x012   
4      Christina   Schwartz             Chiropractor        265-829-3643   
...          ...        ...                      ...                 ...   
54634     Lauren      Reese           Radio producer  977.877.5272x11382   
54635    Derrick      Moore                 Musician        358.661.5483   
54636     Monica  Gutierrez         Graphic designer    294.805.9100x339   
54637      Wanda    Solomon             Toxicologist       (311)767-4924   
54638     Dwayne       Bell  Chief Financial Officer   606-619-7667x9508   

                                     email  client_id

In [148]:
# Create a column for the total price using the subtotal and the shipping price along with a sales tax of 9.25%
item_price = df1['subtotal'] * df1['shipping_price']
price_with_tax =  item_price * .0925
df1['total_price'] = price_with_tax
df1


first       last                      job               phone  \
0         Donald    Harding             Immunologist  793-904-7725x39308   
1        Tiffany      Myers          Music therapist    201.442.4543x942   
2        Shannon     Watson             Immunologist   687.737.9424x8503   
3         Nathan      Baker    Accounting technician    827-788-8123x012   
4      Christina   Schwartz             Chiropractor        265-829-3643   
...          ...        ...                      ...                 ...   
54634     Lauren      Reese           Radio producer  977.877.5272x11382   
54635    Derrick      Moore                 Musician        358.661.5483   
54636     Monica  Gutierrez         Graphic designer    294.805.9100x339   
54637      Wanda    Solomon             Toxicologist       (311)767-4924   
54638     Dwayne       Bell  Chief Financial Officer   606-619-7667x9508   

                                     email  client_id  order_id  order_date  \
0         harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1        myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                    swatson8146@payne.net      57113   1902144  2023-01-29   
3                   bakernathan@benson.com      46554   9031802  2023-04-25   
4      christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   
...                                    ...        ...       ...         ...   
54634   late_reese_4081@montoya-chavez.org      29043   9021716  2023-04-26   
54635         derrick.moore.2602@pope.info      41908   6290153  2023-02-08   
54636            gutierrezm3195@morris.org      35176   8692622  2023-03-05   
54637            solomonwanda5962@ross.org      24485   7592730  2023-01-18   
54638                     belld@galvan.com      24007   7489403  2023-01-04   

       order_week  order_year  ... unit_price unit_cost unit_weight   qty  \
0              17        2023  ...    1096.80    762.71        7.50   105   
1              20        2023  ...      24.95     15.09        1.49    21   
2               4        2023  ...      13.52      7.86        1.68    39   
3              17        2023  ...      36.42     24.85        1.23    29   
4              21        2023  ...     195.10    108.17       46.43    20   
...           ...         ...  ...        ...       ...         ...   ...   
54634          17        2023  ...      83.13     51.60        2.25    33   
54635           6        2023  ...     206.59    175.46       11.70    47   
54636           9        2023  ...      65.66     57.31        4.16   475   
54637           3        2023  ...       1.48      1.22       18.04   112   
54638           1        2023  ...       3.01      1.61        2.07  1031   

       line_number   subtotal  shipping_price    total_price  item_cost  \
0                1  115164.00            75.0  798950.250000   80159.55   
1                0     523.95            14.9     722.134087     331.79   
2                6     527.28            16.8     819.393120     323.34   
3                3    1056.18            12.3    1201.668795     732.95   
4                1    3902.00           464.3  167582.120500    2627.70   
...            ...        ...             ...            ...        ...   
54634            8    2743.29            22.5    5709.472313    1725.30   
54635            0    9709.73           117.0  105083.552925    8363.62   
54636            6   31188.50            41.6  120013.348000   27263.85   
54637            0     165.76           180.4    2766.037120     317.04   
54638            5    3103.31            20.7    5942.062823    1680.61   

       profit  
0        $nan  
1        $nan  
2        $nan  
3        $nan  
4        $nan  
...       ...  
54634    $nan  
54635    $nan  
54636    $nan  
54637    $nan  
54638    $nan  

[54639 rows x 23 columns]

In [173]:
# Create a column for the cost of each line using unit cost, qty, and
# shipping price (assume the shipping cost is exactly what is charged to the client).
ops_cost = order_info['unit_cost'] * order_info['qty']
df1['item_cost'] = ops_cost + df1['shipping_price']
df1


first       last                      job               phone  \
0         Donald    Harding             Immunologist  793-904-7725x39308   
1        Tiffany      Myers          Music therapist    201.442.4543x942   
2        Shannon     Watson             Immunologist   687.737.9424x8503   
3         Nathan      Baker    Accounting technician    827-788-8123x012   
4      Christina   Schwartz             Chiropractor        265-829-3643   
...          ...        ...                      ...                 ...   
54634     Lauren      Reese           Radio producer  977.877.5272x11382   
54635    Derrick      Moore                 Musician        358.661.5483   
54636     Monica  Gutierrez         Graphic designer    294.805.9100x339   
54637      Wanda    Solomon             Toxicologist       (311)767-4924   
54638     Dwayne       Bell  Chief Financial Officer   606-619-7667x9508   

                                     email  client_id  order_id  order_date  \
0         harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1        myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                    swatson8146@payne.net      57113   1902144  2023-01-29   
3                   bakernathan@benson.com      46554   9031802  2023-04-25   
4      christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   
...                                    ...        ...       ...         ...   
54634   late_reese_4081@montoya-chavez.org      29043   9021716  2023-04-26   
54635         derrick.moore.2602@pope.info      41908   6290153  2023-02-08   
54636            gutierrezm3195@morris.org      35176   8692622  2023-03-05   
54637            solomonwanda5962@ross.org      24485   7592730  2023-01-18   
54638                     belld@galvan.com      24007   7489403  2023-01-04   

       order_week  order_year  ... unit_price unit_cost unit_weight   qty  \
0              17        2023  ...    1096.80    762.71        7.50   105   
1              20        2023  ...      24.95     15.09        1.49    21   
2               4        2023  ...      13.52      7.86        1.68    39   
3              17        2023  ...      36.42     24.85        1.23    29   
4              21        2023  ...     195.10    108.17       46.43    20   
...           ...         ...  ...        ...       ...         ...   ...   
54634          17        2023  ...      83.13     51.60        2.25    33   
54635           6        2023  ...     206.59    175.46       11.70    47   
54636           9        2023  ...      65.66     57.31        4.16   475   
54637           3        2023  ...       1.48      1.22       18.04   112   
54638           1        2023  ...       3.01      1.61        2.07  1031   

       line_number   subtotal  shipping_price    total_price  item_cost  \
0                1  115164.00            75.0  798950.250000   80159.55   
1                0     523.95            14.9     722.134087     331.79   
2                6     527.28            16.8     819.393120     323.34   
3                3    1056.18            12.3    1201.668795     732.95   
4                1    3902.00           464.3  167582.120500    2627.70   
...            ...        ...             ...            ...        ...   
54634            8    2743.29            22.5    5709.472313    1725.30   
54635            0    9709.73           117.0  105083.552925    8363.62   
54636            6   31188.50            41.6  120013.348000   27263.85   
54637            0     165.76           180.4    2766.037120     317.04   
54638            5    3103.31            20.7    5942.062823    1680.61   

            profit  
0      $718,790.70  
1          $390.34  
2          $496.05  
3          $468.72  
4      $164,954.42  
...            ...  
54634    $3,984.17  
54635   $96,719.93  
54636   $92,749.50  
54637    $2,449.00  
54638    $4,261.45  

[54639 rows x 23 columns]

In [177]:
# Create a column for the profit of each line using line cost and line price

df1['profit'] = df1['total_price'] - df1['item_cost']
df1['profit'] = df1['profit'].apply(lambda x: f'${x:,.2f}')
df1['order_id'].sort_index


<bound method Series.sort_index of 0        8953482
1        8069089
2        1902144
3        9031802
4        1322274
          ...   
54634    9021716
54635    6290153
54636    8692622
54637    7592730
54638    7489403
Name: order_id, Length: 54639, dtype: int64>

## Part 3: Confirm your work
You have email receipts showing that the total prices for 3 orders. Confirm that your calculations match the receipts. Remember, each order has multiple lines.

Order ID 2742071 had a total price of \$152,811.89

Order ID 2173913 had a total price of \$162,388.71

Order ID 6128929 had a total price of \$923,441.25


In [180]:
# Check your work using the totals above
receipt_order = {
    'order_id': [2742071, 2173913, 6128929],
    'reciept_total_price': [15281189, 16238871, 92344125]

}

confirm_order = df1[order_info['order_id'] == 2742071]
confirm_order

first   last                                       job  \
3446   Kimberly  Blair  Geographical information systems officer   
6287   Kimberly  Blair  Geographical information systems officer   
20623  Kimberly  Blair  Geographical information systems officer   
23840  Kimberly  Blair  Geographical information systems officer   
24671  Kimberly  Blair  Geographical information systems officer   
42470  Kimberly  Blair  Geographical information systems officer   

                    phone                           email  client_id  \
3446   477-680-9323x27378  blair.kimberly.6618@nelson.biz      78447   
6287   477-680-9323x27378  blair.kimberly.6618@nelson.biz      78447   
20623  477-680-9323x27378  blair.kimberly.6618@nelson.biz      78447   
23840  477-680-9323x27378  blair.kimberly.6618@nelson.biz      78447   
24671  477-680-9323x27378  blair.kimberly.6618@nelson.biz      78447   
42470  477-680-9323x27378  blair.kimberly.6618@nelson.biz      78447   

       order_id  order_date  order_week  order_year  ... unit_price unit_cost  \
3446    2742071  2023-02-27           9        2023  ...       5.59      3.05   
6287    2742071  2023-02-27           9        2023  ...     460.20    407.68   
20623   2742071  2023-02-27           9        2023  ...     135.12     80.78   
23840   2742071  2023-02-27           9        2023  ...      47.59     35.41   
24671   2742071  2023-02-27           9        2023  ...      59.69     54.86   
42470   2742071  2023-02-27           9        2023  ...      82.18     61.07   

      unit_weight  qty  line_number   subtotal  shipping_price    total_price  \
3446        10.46   77            5     430.43           104.6    4164.625465   
6287         2.60  260            0  119652.00            26.0  287763.060000   
20623        2.27   12            3    1621.44            22.7    3404.618640   
23840        0.54   22            1    1046.98             5.4     522.966510   
24671        1.57   24            2    1432.56            15.7    2080.435260   
42470        3.04   44            4    3615.92            30.4   10167.967040   

       item_cost       profit  
3446      339.45    $3,825.18  
6287   106022.80  $181,740.26  
20623     992.06    $2,412.56  
23840     784.42     $-261.45  
24671    1332.34      $748.10  
42470    2717.48    $7,450.49  

[6 rows x 23 columns]

## Part 4: Summarize and Analyze
Use the new columns with confirmed values to find the following information.

In [188]:
# How much did each of the top 5 clients by quantity spend? Check your work from Part 1 for client ids.
amt_spent = []
for client in list_top_fiveclients:
    client_data = df1.loc[df1['client_id'] == client]
    total_spent = client_data['amount_spent'].sum()
    amt_spent.append({'client_id': client, 'total_spent': total_spent})

KeyError: 'amount_spent'

In [22]:
# Create a summary DataFrame showing the totals for the for the top 5 clients with the following information:
# total units purchased, total shipping price, total revenue, and total profit. 


In [23]:
# Format the data and rename the columns to names suitable for presentation.

# Define the money columns. 


# Define a function that converts a dollar amount to millions.


# Apply the currency_format_millions function to only the money columns. 


# Rename the columns to reflect the change in the money format. 




In [24]:
# Sort the updated data by "Total Profit (millions)" form highest to lowest and assign the sort to a new DatFrame.
